src/QMCWaveFunctions/lcao/SoaAtomicBasisSet.h

```diff
@@ -146,6 +146,17 @@ struct SoaAtomicBasisSet
     T r_new;
     // T psi_new, dpsi_x_new, dpsi_y_new, dpsi_z_new,d2psi_new;
 
+    PosType Tvec;
```

Comment:
What is this variable?  Comment please

Reply:
Tvec is a translation vector. gendisp is the difference between the ion position and the electron position, while dr is the difference between the nearest image ion (relative to electron) and the electron position.

In order to reduce the number of images that need to be summed over when generating the AO,  Tvec is constructed such that it can be subtracted from the electron position to bring the electron closer to the primitive cell (Tvec is then necessarily added to the phase afterwards).
As it is coded, it actually brings the electron into a box defined by x,y,z-projections of the sum of the prim lattice vectors. This was easier to implement than bring the electron strictly into the primitive cell and also seemed to sufficiently bring down the number of images.

Tvec is not needed if there is already something floating around that tells us which cell the electron is in -- I was not sure if there was anything like this already.

Reply:
This explanation can be put in comments

Reply:
addressed